fvm/evm/types/events.go

```diff
@@ -26,6 +31,72 @@ type Event struct {
 	Payload EventPayload
 }
 
+var _ common.Location = EVMLocation{}
+
+type EVMLocation struct{}
+
+func (l EVMLocation) TypeID(memoryGauge common.MemoryGauge, qualifiedIdentifier string) common.TypeID {
+	id := fmt.Sprintf("%s%s%s", evmLocationPrefix, locationDivider, qualifiedIdentifier)
+	common.UseMemory(memoryGauge, common.NewRawStringMemoryUsage(len(id)))
+
+	return common.TypeID(id)
+}
+
+func (l EVMLocation) QualifiedIdentifier(typeID common.TypeID) string {
+	pieces := strings.SplitN(string(typeID), locationDivider, 2)
+
+	if len(pieces) < 2 {
+		return ""
+	}
+
+	return pieces[1]
+}
+
+func (l EVMLocation) String() string {
+	return evmLocationPrefix
+}
+
+func (l EVMLocation) Description() string {
+	return evmLocationPrefix
+}
+
+func (l EVMLocation) ID() string {
+	return evmLocationPrefix
+}
+
+func (l EVMLocation) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Type string
+	}{
+		Type: "EVMLocation",
+	})
+}
+
+func init() {
+	common.RegisterTypeIDDecoder(
+		evmLocationPrefix,
+		func(_ common.MemoryGauge, typeID string) (common.Location, string, error) {
```

Comment:
Do we need a test for this part specifically?

Reply:
This is covered in the test I wrote, however edge cases with errors are not.